src/ompl/geometric/planners/bitstar/datastructures/IntegratedQueue.h

```diff
@@ -0,0 +1,453 @@
+*  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+*  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+*  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+*  POSSIBILITY OF SUCH DAMAGE.
+*********************************************************************/
+
+/* Authors: Jonathan Gammell */
+
+#ifndef OMPL_GEOMETRIC_PLANNERS_BITSTAR_DATASTRUCTURES_INTEGRATEDQUEUE_
+#define OMPL_GEOMETRIC_PLANNERS_BITSTAR_DATASTRUCTURES_INTEGRATEDQUEUE_
+
+//STL/Boost/etc.:
+//std::pair
+#include <utility>
+//std::vector
+#include <vector>
+//std::list
+#include <list>
+//std::multimap
+#include <map>
+//boost::unordered_map (pre-C++11 std::unordered_map)
+#include <boost/unordered_map.hpp>
+//For boost::function
+#include <boost/function.hpp>
+
+//OMPL:
+//The cost class:
+#include "ompl/base/Cost.h"
+//The optimization objective class:
+#include "ompl/base/OptimizationObjective.h"
+//The nearest neighbours structure
+#include "ompl/datastructures/NearestNeighbors.h"
+
+//BIT*:
+//I am member class of the BITstar class, so I need to include it's definition to be aware of the class BITstar. It has a forward declaration to me.
+#include "ompl/geometric/planners/bitstar/BITstar.h"
+//The vertex class:
+#include "ompl/geometric/planners/bitstar/datastructures/Vertex.h"
+
+namespace ompl
+{
+    namespace geometric
+    {
+
+
+        /** @anchor IntegratedQueue
+        \par Short Description
+        An integrated two-stage queue that consists of vertices expanded into edges to be processed.
+        The integrated queue consists of a vertex expansion queue and an edge processing queue.
+        Vertices are expanded as needed from the vertex queue into edges places in the edge queue.
+        Edges are removed from the edge queue for processing by BIT*.
+        The vertex queue is implemented as a static ordered list of the vertices in the graph with a token (i.e., an iterator)
+        pointing to the next vertex that needs to be expanded. This is specifically a multimap ordered on ompl::base::Cost
+        The edge queue is implemented as an ordered list of potential edges. It is filled by the vertex queue and emptied by
+        popping the best value off the front. It is specifically a multimap ordered on std::pair<ompl::base::Cost, ompl::base::Cost>
+
+        @par Notes:
+            - An eraseEdge() function could be made by mimicking the vertex -> vertexQueue_::iterator lookup datastructure for the edgeQueue_
+        */
+
+        /** \brief A queue of edges to be processed that integrates both the expansion of \ref gVertex "Vertices" and the ordering of the resulting edges. */
+        class BITstar::IntegratedQueue
+        {
+        public:
+            ////////////////////////////////
+            //Data typedefs:
+            /** \brief A typedef for a pair of vertices, i.e., an edge */
+            typedef std::pair<VertexPtr, VertexPtr> vertex_pair_t;
```

Comment:
For some reason I thought this was the style recommended in the Google C++ Style Guide, but I was incorrect. Fixed.